apps/studio/src/schemas/page.ts

```diff
@@ -44,17 +51,19 @@ export const reorderBlobSchema = z.object({
   ),
 })
 
-export const updatePageSchema = basePageSchema.extend({
-  // NOTE: We allow both to be empty now,
-  // in which case this is a no-op.
-  // We are ok w/ this because it doesn't
-  // incur any db writes
-  parentId: z.number().min(1).optional(),
-  pageName: z.string().min(1).optional(),
-})
-
 export const updatePageBlobSchema = basePageSchema.extend({
-  content: z.string(),
+  content: z.string().transform((value, ctx) => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const parsed = safeJsonParse(value)
+    if (schemaValidator(parsed)) {
+      return parsed
+    }
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "Invalid page content",
```

Comment:
we are trying to abort early right, do we need the FATAL: true flag?

Reply:
it's ok, this will still cause the validation to fail. FATAL just means zod stops validating immediately. No big deal for it to gather all validation errors before returning to the user